src/Juvix/Compiler/Asm/Transformation/TempHeight.hs

```diff
@@ -0,0 +1,76 @@
+module Juvix.Compiler.Asm.Transformation.TempHeight where
+
+import Juvix.Compiler.Asm.Data.Stack
+import Juvix.Compiler.Asm.Transformation.Base
+
+computeFunctionTempHeight ::
+  (Member (Error AsmError) r) =>
+  InfoTable ->
+  FunctionInfo ->
+  Sem r FunctionInfo
+computeFunctionTempHeight tab fi = do
+  ps :: [Command] <- recurseFun sig fi
```

Comment:
This can be done with the simplified recursor `recurseS`. You don't need full memory information & validity checking here -- just stack height info. I'll change it.